term_proposal/alan0201tw/proposal.md

```diff
@@ -41,18 +41,29 @@ And since I have a working (simple, but working!) ray tracer at hand, I can do s
     * stb_image is a single header library, and do not require extra set ups, so it probably won't complicate stuffs
     * ffmpeg can be used as an external tool, and be used with shell scripts, thus not affecting the build chain, so it should be okay, too
 
-### TBD
+### Considering design choices
 
 * Use physics-based shading or not?
 * Is providing surface normal the responsibility of the surface, or should be evaluated by ray-casting?
 
 
 ## Implementation
 
-My first instinct is to implement everything in c++, but having a python interface for setting up scenes sounds quite nice, so I might do that (if I have the time).
+The render engine itslef will be implemented in c++. It will be a standalone application. Scene contents will be controlled through c++ code.
```

Comment:
When you say application, I assume it is an executable, and not a shared object.  If this is true, please make it explicit in the proposal.

I'd like to add that a standalone executable will make it difficult to test and script in Python.  It is OK to make this decision, though, as long as you are fully aware of the consequences.

Reply:
Thank you for pointing out the inaccurate word use. I did mean 'executable' when I used the word 'application'.

Designing this engine as a standalone executable is based on my previous experience on ray tracers. I will surely start this project as a standalone executable, but I might switch it to the form of shared library if I am comfortable with it.